pkg/types/types.go

```diff
@@ -20,3 +24,20 @@ func (v JobID) String() string {
 func (v RunID) String() string {
 	return strconv.FormatUint(uint64(v), 10)
 }
+
+// JobIDFromContext is a helper to get the JobID, this is useful
+// for plugins which need to know which job they are running.
+// Not all context object everywhere have this set, but this is
+// guaranteed to work in TargetManagers, TestSteps and Reporters
+func JobIDFromContext(ctx xcontext.Context) (JobID, bool) {
+	v, ok := ctx.Value("job_id").(JobID)
```

Comment:
You need to define a special type for the key and use it. This way you will guarantee no accidental clashing:
```
type jobIDKey struct{}

ctx = ctx.WithValue(jobIDKey, jobID)
```

Reply:
TIL. Fixed